app/models/collection.rb

```diff
@@ -128,6 +130,11 @@ def to_open_search_collections(document, hits, params, url, for_atom = true, cwi
           add_link_as_child(doc, node, "#{ENV['opensearch_url']}/granules/descriptor_document.xml?collectionConceptId=#{id}&clientId=#{params[:clientId]}", 'application/opensearchdescription+xml', NEW_REL_MAPPING[:search], link_title)
         end
 
+        if is_fedeo && provider_osdd_link.present?
```

Comment:
I'd rather we refactor this whole section. We are doing one of the following,
1. Adding an OSDD link that is derived from a tag in the metadata result.
2. Adding an OSDD link because it is a CWIC provider
3. Adding an OSDD link because it has granules in CMR
4. Not adding an OSDD link

Our code should reflect that set of decisions intuitively.

Reply:
One side effect of this change is that previously, a Fedeo collection would have its Non-CMR OSDD inserted as well as have the two links from what is now the `has_granules` case inserted. Now, only one or the other is inserted. I altered the test to accommodate this fact.

Reply:
Cool.